agent/engine/docker_task_engine.go

```diff
@@ -1138,7 +1143,27 @@ func (engine *DockerTaskEngine) AddTask(task *apitask.Task) {
 		engine.emitTaskEvent(task, err.Error())
 		return
 	}
-
+	if task.IsEBSTaskAttachEnabled() {
+		if ebsCsiDaemonManager, ok := engine.daemonManagers["ebs-csi-driver"]; ok {
+			if engine.ebsCsiDriver == nil {
```

Comment:
|| task was stopped 

TODO when the daemon task stops for a valid reason, we will need to replace it -- specifically here when we are waiting on a daemon-dependent task payload.